src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.Windows.cs

```diff
@@ -82,5 +80,28 @@ internal static unsafe int StringToAnsiString(string s, byte* buffer, int buffer
             buffer[nb] = 0;
             return nb;
         }
+
+        // Returns number of bytes required to convert given string to Ansi string. The return value includes null terminator.
+        internal static unsafe int GetAnsiStringByteCount(ReadOnlySpan<char> chars)
+        {
+            fixed (char* pChars = chars)
+            {
+                return Interop.Kernel32.WideCharToMultiByte(
+                   Interop.Kernel32.CP_ACP, Interop.Kernel32.WC_NO_BEST_FIT_CHARS, pChars, chars.Length, null, 0, IntPtr.Zero, IntPtr.Zero) + 1;
+            }
+        }
+
+        // Converts given string to Ansi string. The converted value includes null terminator.
+        internal static unsafe int GetAnsiStringBytes(ReadOnlySpan<char> chars, Span<byte> bytes)
+        {
+            fixed (char* pChars = chars)
+            fixed (byte* pBytes = bytes)
+            {
+                int actualByteLength = Interop.Kernel32.WideCharToMultiByte(
+                   Interop.Kernel32.CP_ACP, Interop.Kernel32.WC_NO_BEST_FIT_CHARS, pChars, chars.Length, pBytes, bytes.Length, IntPtr.Zero, IntPtr.Zero);
+                bytes[actualByteLength] = 0;
```

Comment:
WideCharToMultiByte

Reply:
The caller is expected provide buffer that is big enough for the converted string, including the null terminator. If the buffer provided by the caller is not large enough (e.g. due to bug), this is going to throw IndexOutOfRangeException.

(The interop code often makes fragile assumptions like this. It won't hurt to harden it using Span.)

Reply:
Added comment.